src/components/admin/article/ArticleController.jsx

```diff
@@ -60,35 +64,31 @@ class ArticleController extends FalcorController {
     }, 500);
   }
 
-  save(jsonGraphEnvelope, processedAuthors, articleSlug, falcorData) {
+  async save(jsonGraphEnvelope, processedAuthors, articleSlug, falcorData) {
     // Update the values
     this.safeSetState({ saving: true });
-    const updatePromises = [this.falcorUpdate(jsonGraphEnvelope)];
+    await this.falcorUpdate(jsonGraphEnvelope);
```

Comment:
Could we maybe do a check whether a slug has changed and if not then call both of these at the same time with `Promise.all`? The reason I want to do this is that on your local machine the save probably is basically instant, but in production with internet latency it actually takes 1-2 seconds, and this would double that save time.

Also I'd like to chat with you about the URL changing etc. as it's not something I've thought about before with the slug changing

Reply:
This is actually why I had it set to not `await` for the second falcorCall. There is no reason to wait for the `updateAuthors` call to complete. If you are worried about time, it makes more sense to do that as it makes the client speed faster even if we change the slug.

Reply:
I had forgotten that was why I did that... My brain needs to be running fewer threads I guess haha

Reply:
Well aren't there actually also some race conditions going on with changing the slug in the database, since it'll try to change the authors of a slug that no longer exists? 

And also, it'll run straight down to the setState call that says save was complete, even if the authors haven't been updated yet

Reply:
`this.state.data` is updated when `falcorCall` returns, so yes it should.

I see what you mean by the await handling it, but why not just put it in a `Promise.all` then instead of a non waited upon promise, I think that's much clearer and also correct. In case it fails `falcorController` does right now throw a `window.alert` so it would be weird if the save was successful and then afterwards an alert with the error pops up

Reply:
I actually have a way I am okay with implementing promise.all. Rather than conditionally using `Promise.all` I am going to separate the updating of slugs into it's own Promise and then batch the rest of the changes with `Promise.all`. I think that is a solution that you'll be happier with :)

Reply:
And then conditionally run the slug updating promise? Yes I like that better :D. Thanks for accommodating me ;)